### PR TITLE
Add helper for parsing context from the encoded metric traces

### LINKED ISSUE #777: Add "object" decoding to to `encoding.Decoder`, then update `tests/integration/golang/aim/metric/search_test.go` to use it

### PR DESCRIPTION
**Pull request by suprjinx**
fixes #777

### CODE REVIEW

File: tests/integration/golang/aim/metric/search_test.go [hunk 1]
```diff
@@ -162,11 +162,7 @@ func (s *SearchMetricsTestSuite) Test_Ok() {
 		RunID:     run2.ID,
 		Iter:      3,
 		Context: models.Context{
-			Json: datatypes.JSON([]byte(`
-				{
-					"testkey": "testvalue"
-				}`,
-			)),
+			Json: datatypes.JSON([]byte(`{"testkey":"testvalue"}`)),
```
Comment (dsuhinin) [inline tests/integration/golang/aim/metric/search_test.go:165]: we probably don't need to have `datatypes.JSON([]byte`. under the hood `datatypes.JSON` is already `[]byte` so we can use just `datatypes.JSON` without `[]byte`.

File: tests/integration/golang/aim/metric/search_test.go [hunk 2]
```diff
@@ -179,11 +175,7 @@ func (s *SearchMetricsTestSuite) Test_Ok() {
 		RunID:     run2.ID,
 		LastIter:  3,
 		Context: models.Context{
-			Json: datatypes.JSON([]byte(`
-				{
-					"testkey": "testvalue"
-				}`,
-			)),
+			Json: datatypes.JSON([]byte(`{"testkey":"testvalue"}`)),
```
Comment (dsuhinin) [inline tests/integration/golang/aim/metric/search_test.go:178]: the same probably here.

File: tests/integration/golang/aim/metric/search_test.go [hunk 3]
```diff
@@ -5929,6 +5921,16 @@ func (s *SearchMetricsTestSuite) Test_Ok() {
 					timestampsKey := prefix + ".timestamps.blob"
 					valuesKey := prefix + ".values.blob"
 
+					contextPrefix := prefix + ".context"
+					contx, err := helpers.ExtractContextBytes(contextPrefix, decodedData)
+					s.Require().Nil(err)
```
Comment (dsuhinin) [inline tests/integration/golang/aim/metric/search_test.go:5926]: I think it should be a part of our standard  decoder unfortunately and work for any object and not only for `context`.

File: tests/integration/golang/helpers/context.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+// ExtractContextBytes finds the metric context in the decoded data, using the given key prefix.
+func ExtractContext(contextPrefix string, decodedData map[string]any) map[string]any {
+	contx := map[string]any{}
+	for key := range decodedData {
+		if strings.HasPrefix(key, contextPrefix) && len(key) > len(contextPrefix) {
+			contx[key[len(contextPrefix)+1:]] = decodedData[key]
+		}
+	}
+	return contx
+}
```
Comment (dsuhinin) [inline tests/integration/golang/helpers/context.go:24]: Im not sure that this is what we exactly need. so:
- extraction should be a part of our `decoder`.
- extraction should work for any object and not only for context out of the box.
- `json` object could be complex object with other nested objects and extraction should work for this case as well under the hood of `decoder`.

**Review by dsuhinin — CHANGES_REQUESTED** [review]

**Review by dsuhinin — DISMISSED** [review]

File: tests/integration/golang/aim/metric/search_test.go (hunk 1 above)
Comment (suprjinx) [reply tests/integration/golang/aim/metric/search_test.go:165]: done

File: tests/integration/golang/aim/metric/search_test.go (hunk 3 above)
Comment (suprjinx) [reply tests/integration/golang/aim/metric/search_test.go:5926]: done

File: tests/integration/golang/helpers/context.go (hunk 4 above)
Comment (suprjinx) [reply tests/integration/golang/helpers/context.go:24]: saving this improvement for another day -- possibly conversion to arrow records will happen first.

File: tests/integration/golang/aim/metric/search_test.go (hunk 2 above)
Comment (suprjinx) [reply tests/integration/golang/aim/metric/search_test.go:178]: done

**Review by dsuhinin — APPROVED** [review]

**Pull request merged**